src/EditorFeatures/Core/Implementation/IntelliSense/QuickInfo/Presentation/QuickInfoDisplayPanel.cs

```diff
@@ -1,77 +1,103 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
 using System.Text;
 using System.Windows;
 using System.Windows.Controls;
 using System.Windows.Documents;
 using System.Windows.Media;
+using Microsoft.CodeAnalysis.QuickInfo;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.QuickInfo
 {
+    internal class TextBlockElement
+    {
+        public string Kind { get; }
+        public TextBlock Block { get; }
+
+        public TextBlockElement(string kind, TextBlock block)
+        {
+            this.Kind = kind;
+            this.Block = block;
+        }
+    }
+
     internal class QuickInfoDisplayPanel : StackPanel
     {
-        internal TextBlock MainDescription { get; }
-        internal TextBlock Documentation { get; }
-        internal TextBlock TypeParameterMap { get; }
-        internal TextBlock AnonymousTypes { get; }
-        internal TextBlock UsageText { get; }
-        internal TextBlock ExceptionText { get; }
+        private ImmutableArray<TextBlockElement> _textBlocks;
+
+        internal TextBlock MainDescription => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.Description)?.Block;
+        internal TextBlock Documentation => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.DocumentationComments)?.Block;
+        internal TextBlock TypeParameterMap => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.TypeParameters)?.Block;
+        internal TextBlock AnonymousTypes => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.AnonymousTypes)?.Block;
+        internal TextBlock UsageText => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.Usage)?.Block;
+        internal TextBlock ExceptionText => _textBlocks.FirstOrDefault(tb => tb.Kind == QuickInfoTextKinds.Exception)?.Block;
```

Comment:
So, it's not totally clear to me why these are needed.  Do we have specfici code elsewhere that needs to know "is this the type parameter map"?  It seems highly desirable for "private ImmutableArray<TextBlockElement> _textBlocks;" to contain enough information in it alone that the presentation can work off of.